### DISCUSSION TITLE
启动loki插件失败

### DISCUSSION THREAD

**Discussion by kingzyt** [kubevela/kubevela#6026] [Q&A] [answered]
我一开始是成功安装了loki到local的，但后来新增了一个trial集群，但这个集群开始访问有点问题，造成loki在部署到这个集群的过程中失败了，然后我用一个新的集群替换了原来不可用的集群，名字还是用了trial，但loki还是无法启动：
在vela status addon-loki -n vela-system中看到在export-service阶段报错：
    name: export-service
    type: export-service
    phase: failed
    message: run step(provider=kube,do=apply): pre-dispatch dryrun failed: Found 1 errors. [(cannot create object: Endpoints "loki" is invalid: [subsets[0].addresses[0].ip: Invalid value: "loki.o11y-system": must be a valid IP address, (e.g. 10.9.8.7 or 2001:db8::ffff), subsets[0].addresses[0].ip: Invalid value: "loki.o11y-system": must be a valid IP address])]
因为有使用了traits的file-log，所以系统不给删除，现在就卡着了，不知道如何修复

另外vela在添加新集群后，貌似也不会主动同步插件到新集群，需要手动一个个addon吗？

**Comment by Somefive** [comment, marked as answer]
1. 你需要安装 loki 到子集群嘛？还是说只需要在管控中使用呢？
2. KubeVela 目前默认的 Loki 组件在多集群部署的情况下并非是每个集群部署一个 Loki，而是管控集群部署一个 Loki，子集群部署 Agent (Promtail 或 Vector)，在多集群部署的情况下需要把 vela addon enable serviceType=LoadBalancer 使得不同集群的 agent 和 loki 能够跨集群通信。
